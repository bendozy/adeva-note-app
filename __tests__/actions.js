import configureMockStore from 'redux-mock-store';
import {
  createNote,
  updateNote,
  deleteNote,
} from '../src/actions/notes';

const mockStore = configureMockStore();

describe('Note action creators tests', () => {
  it('creates CREATE_NOTE when creating a note is successful', () => {
    const store = mockStore({});

    store.dispatch(createNote({
      title: 'Example',
      details: 'Testing detail',
    }));
    expect(store.getActions()).toMatchSnapshot();
  });

  it('creates UPDATE_NOTE when updating a note is successful', () => {
    const initialState = {
      notes: [{
        id: 2,
        title: 'Example',
        details: 'Testing detail',
      }],
    };
    const store = mockStore(initialState);

    store.dispatch(updateNote({
      id: 2,
      title: 'Update Example',
      details: 'Testing detail updated',
    }));
    expect(store.getActions()).toMatchSnapshot();
  });

  it('creates DELETE_NOTE when deleting a note is successful', () => {
    const initialState = {
      notes: [{
        id: 3,
        title: 'Delete Example',
        details: 'Testing detail',
      }],
    };
    const store = mockStore(initialState);

    store.dispatch(deleteNote(3));
    expect(store.getActions()).toMatchSnapshot();
  });
});
