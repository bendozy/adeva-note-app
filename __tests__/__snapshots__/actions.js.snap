// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note action creators tests creates CREATE_NOTE when creating a note is successful 1`] = `
Array [
  Object {
    "note": Object {
      "details": "Testing detail",
      "title": "Example",
    },
    "type": "CREATE_NOTE",
  },
]
`;

exports[`Note action creators tests creates DELETE_NOTE when deleting a note is successful 1`] = `
Array [
  Object {
    "id": 3,
    "type": "DELETE_NOTE",
  },
]
`;

exports[`Note action creators tests creates UPDATE_NOTE when updating a note is successful 1`] = `
Array [
  Object {
    "note": Object {
      "details": "Testing detail updated",
      "id": 2,
      "title": "Update Example",
    },
    "type": "UPDATE_NOTE",
  },
]
`;
